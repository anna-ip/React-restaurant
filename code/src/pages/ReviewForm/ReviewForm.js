import React, { useState, useContext } from "react";
import { ReviewsContext } from "../../contexts/ReviewsContext";
import styles from "./ReviewForm.module.scss";

const ReviewForm = () => {
  const [name, setName] = useState("");
  const [text, setText] = useState("");
  const { addReview } = useContext(ReviewsContext);

  const timeElapsed = Date.now();
  const today = new Date(timeElapsed).toDateString();

  const handleSubmit = (e) => {
    e.preventDefault();
    // add Star rating as well
    addReview(today, name, text);

    // clears form after submit
    setName("");
    setText("");
  };

  return (
    <div className={styles.container}>
      <form onSubmit={handleSubmit} className={styles.form}>
        <p>{today}</p>
        <label>
          Name:
          <input
            type="text"
            name="name"
            placeholder="name"
            value={name}
            onChange={(e) => setName(e.target.value)}
          />
        </label>
        <p>
          <span role="img" aria-label="rating-stars">
            ⭐︎⭐︎⭐︎⭐︎⭐︎
          </span>
        </p>
        <label>
          Review:
          <textarea
            type="textarea"
            name="reviewText"
            placeholder="Your review"
            value={text}
            rows="5"
            onChange={(e) => setText(e.target.value)}
          />
        </label>
        <input type="submit" value="Post review" className={styles.submitBtn} />
      </form>
    </div>
  );
};

export default ReviewForm;
