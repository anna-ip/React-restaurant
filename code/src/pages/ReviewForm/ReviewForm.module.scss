@import '../../styles/_variables';

.container{
  display: flex;
  justify-content: center;
  align-items: center;
  height: 600px;
  background-image: url(../../images/pexels-emre-can-2079438.jpg);
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  margin-bottom: 2rem;

    .form {
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: space-evenly;
      width: 80%;
      height: 90%;
      background-color: $turquise-color;
      opacity: 0.9;

      label {
        display: flex;
        flex-direction: column;
      }

      .submitBtn{
        width: 8rem;
        height: 2rem;
        text-transform: uppercase;
        background: transparent;
        cursor: pointer;
        align-self: center;
        border: 2px solid $red-color;
        color: $red-color;
        border-radius: 5px;

        &:hover {
          background:$red-color;
          color: $base-color;
        }
      }
    }
}