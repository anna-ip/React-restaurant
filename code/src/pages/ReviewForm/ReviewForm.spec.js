import React from "react";
import { mount, shallow } from "enzyme";
import ReviewForm from "./ReviewForm";

describe("<ReviewForm/>", () => {
  test("render correctly", () => {
    const wrapper = shallow(<ReviewForm />);
    expect(wrapper).toMatchSnapshot();
  });

  //gives TypeError:addReview is not a function
  test.skip("when form is submitted the event is cancelled", () => {
    const wrapper = mount(<ReviewForm />);
    let prevented = false;
    wrapper.find("form").simulate("submit", {
      preventDefault: () => {
        prevented = true;
      },
    });
    expect(prevented).toBe(true);
  });

  //TypeError: addReview is not a function
  test.skip("calls onSubmit prop function when form is submitted", () => {
    const onSubmitFn = jest.fn();
    const wrapper = shallow(<ReviewForm onSubmit={onSubmitFn} />);

    let prevented = false;

    const form = wrapper.find("form");
    form.simulate("submit", {
      preventDefault: () => {
        prevented = true;
      },
    });
    expect(prevented).toBe(true);
    expect(onSubmitFn).toHaveBeenCalledTimes(1);
  });
});
