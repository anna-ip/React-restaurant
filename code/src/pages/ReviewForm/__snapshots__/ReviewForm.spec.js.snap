// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReviewForm/> render correctly 1`] = `
<div
  className="container"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <p>
      Fri Jun 04 2021
    </p>
    <Input
      label="Name:"
      name="name"
      onChange={[Function]}
      placeholder="name"
      required={true}
      type="text"
      value=""
    />
    <StarRating
      handleRating={[Function]}
      onChange={[Function]}
      rating={0}
    />
    <TextArea
      label="Review:"
      name="reviewText"
      onChange={[Function]}
      placeholder="Your review"
      required={true}
      rows="5"
      value=""
    />
    <Input
      className="btn"
      type="submit"
      value="Post review"
    />
    <Button
      className="btn"
      onClick={[Function]}
      value="Take me back"
    />
  </form>
</div>
`;
